src/main/java/net/minecraftforge/event/datafix/ConfigureDataFixSchemaEvent.java

```diff
@@ -0,0 +1,208 @@
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.datafix;
+
+import com.google.common.collect.Maps;
+import com.mojang.datafixers.DSL;
+import com.mojang.datafixers.schemas.Schema;
+import com.mojang.datafixers.types.templates.TypeTemplate;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.datafix.fixes.References;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.fml.event.IModBusEvent;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Event fired in parallel on the mod event bus to allow for the modification
+ * of a DFU Schemas type signature.
+ *
+ * If used to register a new entity or block entity to a particular version,
+ * then this also will need to be registered to all newer versions to persist properly!
+ * As opposed to vanilla which automatically transfers the entities and block entities from older
+ * schemas to newer ones, this event does not do that and the modder is on his own to implement this properly.
+ *
+ * We suggest instead of checking for version equality to check for a greater or equal version.
+ */
+public class ConfigureDataFixSchemaEvent extends Event implements IModBusEvent
+{
+
+    private final Schema schema;
+
+    private final Map<String, Supplier<TypeTemplate>> entityTypes = Maps.newConcurrentMap();
+    private final Map<String, Supplier<TypeTemplate>> blockEntityTypes = Maps.newConcurrentMap();
+
+    //Some unmodifiable views of the above maps
+    private final Map<String, Supplier<TypeTemplate>> entityTypesView = Collections.unmodifiableMap(entityTypes);
+    private final Map<String, Supplier<TypeTemplate>> blockEntityTypesView = Collections.unmodifiableMap(blockEntityTypes);
+
+    public ConfigureDataFixSchemaEvent(final Schema schema)
+    {
+        this.schema = schema;
+    }
+
+    /**
+     * The schema which you can configure.
+     *
+     * @return The schema
+     */
+    public Schema getSchema()
+    {
+        return schema;
+    }
+
+    /**
+     * The version of the schema that is about to be configured.
+     *
+     * @return The version.
+     */
+    public int getVersion()
+    {
+        return getSchema().getVersionKey();
+    }
+
+    /**
+     * Registers a new simple (catch-all-nbt) block entity to the current schema and all its descendent versions.
+     *
+     * @param registryName The registry name of the block entity.
+     */
+    public void registerSimpleBlockEntity(final ResourceLocation registryName)
+    {
+        registerBlockEntity(registryName, DSL::remainder);
+    }
+
+    /**
+     * Registers a new simple (catch-all-nbt) block entity with an item inventory array on the nbt field with the given name to the current schema and all its descendent versions.
```

Comment:
Should this be block entity type instead of just block entity?

Reply:
No this does not refer to the type, but to the NBT of the actual block entity.